### PR TITLE
[CustomProperties] Added test for custom properties

### PR DESCRIPTION
**Pull request by AndrewMusgrave**
### WHY are these changes introduced?

We have incorrect custom properties

### WHAT is this pull request doing?

Adding utilities and tests for custom properties

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}

```

</details>

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

**Comment by AndrewMusgrave** [comment]
Before I start adding tests/polish I was wondering if you have any initial thoughts cc/ @danrosenthal @tmlayton . I noticed we have some incorrect custom properties so I wrote out some utilities and added a test that'll give us more confidence using custom properties moving forward.

**Comment by tmlayton** [comment]
> We have incorrect custom properties

Which ones?

**Comment by AndrewMusgrave** [comment]
They'll fail on [CI now](https://circleci.com/gh/Shopify/polaris-react/37803). Here the [commit](https://github.com/Shopify/polaris-react/pull/2616/commits/c3d641610b22d65a1c7c77d284b0b646845498f4) fixing them.

**Comment by tmlayton** [comment]
> initial thoughts

I like it :)

**Comment by danrosenthal** [comment]
> initial thoughts

> I like it :)

Yes, that!

**Comment by BPScott** [comment]
Loving this.

As an implementation hook - I wonder if we could use postcss and the postcss-scss plugin - which we already use as part of our build, and already have available via sewing-kit to generate the AST to check the custom properties rather than bring in a new dependency?

This utility isn't something we want to distribute so it feels like it should live in the tests folder, like `build.test.js`

**Comment by AndrewMusgrave** [comment]
> As an implementation hook - I wonder if we could use postcss and the postcss-scss plugin - which we already use as part of our build, and already have available via sewing-kit to generate the AST to check the custom properties rather than bring in a new dependency?

One idea I was going to explore next was to make this a module that could be shared between other repo's (polaris-rails, web, core, etc). I'm not too familiar with postcss plugins, could we fail the build if we found incorrect custom properties? What kind of warnings/errors can we surface. And if we wanted to share this to other projects, they would have to use postcss which they may not be using.

> This utility isn't something we want to distribute so it feels like it should live in the tests folder, like build.test.js

100% agree. I placed it in `src/utilities` because I assumed `/tests` didn't transpile typescript code. After a check, it looks like it does, any idea why the build test isn't typescript?

**Comment by BPScott** [comment]
> 100% agree. I placed it in `src/utilities` because I assumed `/tests` didn't transpile typescript code. After a check, it looks like it does, any idea why the build test isn't typescript?

Not a clue. Perhaps it used to not work then at some point we chnaged the TS root directory from `src` to `.`

**Comment by danrosenthal** [comment]
Whoops, I accidentally pushed some of my changes to this branch. ☝️

File: src/utilities/theme/tokens.ts [hunk 1]
```diff
@@ -24,6 +24,7 @@ const Overrides = {
   nonNullContent: "''",
   choiceSize: rem('20px'),
   choiceMargin: rem('1px'),
+  controlBorderWidth: rem('2px'),
```
Comment (AndrewMusgrave) [inline src/utilities/theme/tokens.ts]: Grabbed this from your PR Dan 😄

**Comment by AndrewMusgrave** [comment]
Talked to Dan R. and we're hoping to ship this so we'll have confidence incorrect custom properties won't be used. And I can continue to iterate and add tests to this before we release it as a package that'll be consumed by other libraries during our rollout.

File: .eslintignore [hunk 2]
```diff
@@ -11,3 +11,5 @@ node_modules
 /styles.scss
 /styles
 /src/styles/polaris-tokens
+/scripts/analyze-custom-properties/dist
+custom-properties-data.json
```
Comment (BPScott) [inline .eslintignore]: I'm not a fan of adding new stuff into the root. Can we put this (and analyze-custom-properties/dist is possible) into the build folder?

**Comment by BPScott** [comment]
With `splash` we use babel-node to execute typescript directly instead of needing to an intermediate build step. Yuo can probably do the same here to get rid of a bunch of your build scripting and that dist folder.

Ask @amrocha for more info

**Comment by AndrewMusgrave** [comment]
> With splash we use babel-node to execute typescript directly instead of needing to an intermediate build step. You can probably do the same here to get rid of a bunch of your build scripting and that dist folder.

Most of the build script will be removed when this is a separate package. The other part builds the custom properties we expect. We could just use another runtime for that. I usually use `ts-node`. Is there a reason to prefer `babel-node` or just a preference?

**Comment by BPScott** [comment]
> I usually use ts-node. Is there a reason to prefer babel-node or just a preference?

No strong reason to go one over the other - We're already using babel-node for other stuff so feels like keeping to existing tooling feels handier.

**Pull request merged**